### PR TITLE
Update app metadata restriction fields

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/13973

Follow up the suggestion from @marians, https://github.com/giantswarm/apiextensions/pull/615#issuecomment-725310458 

Spec: https://github.com/giantswarm/giantswarm/blob/master/content/docs/architecture-specs-adrs/specs/managed-apps/2020-05-05-app-versions-representation.md

## Checklist

- [x] Consider SIG UX feedback.
- [ ] ~Update changelog in CHANGELOG.md.~
- [ ] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

File: config/crd/v1/application.giantswarm.io_appcatalogentries.yaml [hunk 1]
```diff
@@ -114,24 +114,20 @@ spec:
...
-                      GPU instances to run.
+                      GPU instances to run. Default is false.
                     type: boolean
                   namespace_singleton:
                     description: NamespaceSingleton is a flag for whether this app
-                      can be installed at most once per namespace.
+                      can be installed at most once per namespace. Default is false.
                     type: boolean
-                required:
-                - cluster_singleton
```
Comment (tomahawk28) [inline config/crd/v1/application.giantswarm.io_appcatalogentries.yaml:132]: Now all fields belong to `restrictions` are not required any longer.

File: config/crd/v1/application.giantswarm.io_appcatalogentries.yaml [hunk 2]
```diff
@@ -114,24 +114,20 @@ spec:
                 properties:
                   cluster_singleton:
                     description: ClusterSingleton is a flag for whether this app can
-                      be installed at most once per cluster.
+                      be installed at most once per cluster. Default is false.
```
Comment (rossf7) [inline config/crd/v1/application.giantswarm.io_appcatalogentries.yaml:117]: Nice!  Much clearer with the defaults explained.

**Review by rossf7 — APPROVED** [review]
LGTM

But good to get a review from the SIG too and @piontec please check when you're back.

**Review by marians — APPROVED** [review]

File: config/crd/v1/application.giantswarm.io_appcatalogentries.yaml (hunk 2 above)
Comment (rossf7) [reply config/crd/v1/application.giantswarm.io_appcatalogentries.yaml:117]: Good catch on the 🐫 case. 👍

Comment (marians) [reply config/crd/v1/application.giantswarm.io_appcatalogentries.yaml:117]: Very good indeed. I didn't notice that.

**Review by marians — APPROVED** [review]

**Comment by tomahawk28** [comment]
@piontec Can you maybe check this again? if its okay, I'm going to merge this

**Pull request merged**